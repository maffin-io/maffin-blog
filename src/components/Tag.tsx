import React from 'react';
import Link from 'next/link';

type TagProps = {
  text: string,
};

export default function Tag({ text }: TagProps): JSX.Element {
  return (
    <Link href={`/tags/${text}`}>
      <span className="mr-3 text-sm font-medium uppercase text-primary-500 hover:text-primary-600 dark:hover:text-primary-400">
        #
        {text}
      </span>
    </Link>
  );
}
