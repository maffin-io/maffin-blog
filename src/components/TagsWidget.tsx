import React from 'react';

import Tag from '@/components/Tag';

export type TagsWidgetProps = {
  tags: string[];
};

export default function TagsWidget({ tags }: TagsWidgetProps): JSX.Element {
  return (
    <div>
      {
        tags
        && !tags.includes('docs')
        && (
          <div className="py-4 xl:py-8">
            <h2 className="text-xs uppercase tracking-wide text-gray-500 dark:text-gray-400">
              Tags
            </h2>
            <div className="flex flex-wrap">
              {tags.map((tag) => (
                <Tag key={tag} text={tag} />
              ))}
            </div>
          </div>
        )
      }
    </div>
  );
}
