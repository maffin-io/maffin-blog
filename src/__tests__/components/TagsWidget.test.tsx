import { render } from '@testing-library/react';

import Tag from '@/components/Tag';
import TagsWidget from '@/components/TagsWidget';

jest.mock('@/components/Tag', () => jest.fn(
  () => <div data-testid="Tag" />,
));

describe('TagsWidget', () => {
  it('is empty when no tags', () => {
    render(<TagsWidget tags={[]} />);
    expect(Tag).toBeCalledTimes(0);
  });

  it('is empty when "docs" in the tags', () => {
    render(<TagsWidget tags={['a', 'b', 'docs']} />);
    expect(Tag).toBeCalledTimes(0);
  });

  it('renders as expected when valid tags', () => {
    const { container } = render(<TagsWidget tags={['a', 'b']} />);
    expect(Tag).toBeCalledTimes(2);
    expect(container).toMatchSnapshot();
  });
});
