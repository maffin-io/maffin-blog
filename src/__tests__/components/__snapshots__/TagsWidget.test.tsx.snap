// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagsWidget renders as expected when valid tags 1`] = `
<div>
  <div>
    <div
      class="py-4 xl:py-8"
    >
      <h2
        class="text-xs uppercase tracking-wide text-gray-500 dark:text-gray-400"
      >
        Tags
      </h2>
      <div
        class="flex flex-wrap"
      >
        <div
          data-testid="Tag"
        />
        <div
          data-testid="Tag"
        />
      </div>
    </div>
  </div>
</div>
`;
